### PR TITLE
Handle primitive types in AvroRecordWriterProvider.

### LINKED ISSUE #130: exception when using StringConverter for key and value converter
I'm working to see if this is fixable.

```
[2016-09-28 19:16:22,211] ERROR Task coinsmith-raw-hdfs-sink-test-8-0 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerSinkTask)
org.apache.avro.file.DataFileWriter$AppendWriteException: org.apache.avro.AvroRuntimeException: Unknown datum type io.confluent.kafka.serializers.NonRecordContainer: io.confluent.kafka.serializers.NonRecordContainer@31d89133
        at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:296)
        at io.confluent.connect.hdfs.avro.AvroRecordWriterProvider$1.write(AvroRecordWriterProvider.java:64)
        at io.confluent.connect.hdfs.avro.AvroRecordWriterProvider$1.write(AvroRecordWriterProvider.java:59)
        at io.confluent.connect.hdfs.TopicPartitionWriter.writeRecord(TopicPartitionWriter.java:487)
        at io.confluent.connect.hdfs.TopicPartitionWriter.write(TopicPartitionWriter.java:264)
        at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:234)
        at io.confluent.connect.hdfs.HdfsSinkTask.put(HdfsSinkTask.java:91)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:381)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:227)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:170)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:142)
        at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:140)
        at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:175)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.avro.AvroRuntimeException: Unknown datum type io.confluent.kafka.serializers.NonRecordContainer: io.confluent.kafka.serializers.NonRecordContainer@31d89133
        at org.apache.avro.generic.GenericData.getSchemaName(GenericData.java:636)
        at org.apache.avro.generic.GenericData.resolveUnion(GenericData.java:601)
        at org.apache.avro.generic.GenericDatumWriter.resolveUnion(GenericDatumWriter.java:151)
        at org.apache.avro.generic.GenericDatumWriter.write(GenericDatumWriter.java:71)
        at org.apache.avro.generic.GenericDatumWriter.write(GenericDatumWriter.java:58)
        at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:290)
        ... 17 more
```

### PR DESCRIPTION
**Pull request by ewencp**
Fixes #130

### CODE REVIEW

**Comment by ewencp** [comment]
@rnpridgeon @confluentinc/connect

**Comment by ewencp** [comment]
This issue has been around for quite awhile, I targeted 3.0.x and we can merge up to 3.1.x, 3.2.x, and master.

File: src/test/java/io/confluent/connect/hdfs/HdfsSinkTaskTest.java [hunk 1]
```diff
@@ -179,6 +180,49 @@ public void testSinkTaskPut() throws Exception {
...
+        long startOffset = validOffsets[j - 1] + 1;
+        long endOffset = validOffsets[j];
+        Path path = new Path(FileUtils.committedFileName(url, topicsDir, directory, tp,
+                startOffset, endOffset, extension,
+                ZERO_PAD_FMT));
+        Collection<Object> records = schemaFileReader.readData(conf, path);
+        long size = endOffset - startOffset + 1;
+        assertEquals(records.size(), size);
+        for (Object avroRecord : records) {
+          assertEquals(avroRecord, 12);
```
Comment (kkonstantine) [inline src/test/java/io/confluent/connect/hdfs/HdfsSinkTaskTest.java]: nit: Use the variable `record` instead? Rename it to something that stresses that it's just a static value?

Comment (ewencp) [inline src/test/java/io/confluent/connect/hdfs/HdfsSinkTaskTest.java]: Ack, cleaned it up a bit.

**Pull request merged**